src/Parser.js

```diff
@@ -764,52 +706,60 @@ export default class Parser {
     }
 
     /**
-     * If `optional` is false or absent, this parses an ordinary group,
-     * which is either a single nucleus (like "x") or an expression
-     * in braces (like "{x+y}") or an implicit group, a group that starts
-     * at the current position, and ends right before a higher explicit
+     * Parses an argument.
+     */
+    parseArgumentGroup(optional: boolean, mode?: Mode): ?AnyParseNode {
+        const argToken = this.gullet.scanArgument(optional);
+        if (argToken == null) {
+            return null;
+        }
+        const outerMode = this.mode;
+        if (mode) { // Switch to specified mode
+            this.switchMode(mode);
+        }
+
+        this.gullet.beginGroup();
+        const expression = this.parseExpression(false, "EOF");
+        this.expect("EOF"); // expect the end of the argument
```

Comment:
I wonder if we could use a different token to mark the end of arguments.  Overloading `EOF` seems like it could be a bit confusing.  Maybe something for a future PR given that there's already so much going on in this one.

Reply:
@kevinbarabash I agree. This is my major concern, too. This will probably cause a lot of issues that are hard to debug, but I can't seem to find a simple better way.